src/main/java/connectfour/GameController.java

```diff
@@ -124,6 +143,62 @@ private Parent sendStorage() throws IOException {
+                        addCirclePawnToGridPaneBoardWin(i, j);
+                        addCirclePawnToGridPaneBoardWin(i + 1, j);
+                        addCirclePawnToGridPaneBoardWin(i + 2, j);
+                        addCirclePawnToGridPaneBoardWin(i + 3, j);
+                    }
+                }
+            }
+        } else if (game.getBoard().hasFourBiasUp(game.getCurrentPlayer().getPawn())) {
+            for (int i = 3; i < 6; i++) {
+                for (int j = 0; j < (7 - 3); j++) {
+                    if (game.getBoard().getFields()[i][j] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i - 1][j + 1] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i - 2][j + 2] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i - 3][j + 3] == game.getCurrentPlayer().getPawn()) {
+                        addCirclePawnToGridPaneBoardWin(i, j);
+                        addCirclePawnToGridPaneBoardWin(i - 1, j + 1);
+                        addCirclePawnToGridPaneBoardWin(i - 2, j + 2);
+                        addCirclePawnToGridPaneBoardWin(i - 3, j + 3);
+                    }
+                }
+            }
+        } else if (game.getBoard().hasFourBiasDown(game.getCurrentPlayer().getPawn())) {
+            for (int i = 0; i < (6 - 3); i++) {
+                for (int j = 0; j < (7 - 3); j++) {
+                    if (game.getBoard().getFields()[i][j] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i + 1][j + 1] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i + 2][j + 2] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i + 3][j + 3] == game.getCurrentPlayer().getPawn()) {
+                        addCirclePawnToGridPaneBoardWin(i, j);
+                        addCirclePawnToGridPaneBoardWin(i + 1, j + 1);
+                        addCirclePawnToGridPaneBoardWin(i + 2, j + 2);
+                        addCirclePawnToGridPaneBoardWin(i + 3, j + 3);
+                    }
+                }
+            }
+        } else if (game.getBoard().hasFourHorizontal(game.getCurrentPlayer().getPawn())) {
+            for (int i = 0; i < (6); i++) {
+                for (int j = 0; j < (7 - 3); j++) {
+                    if (game.getBoard().getFields()[i][j] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i][j + 1] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i][j + 2] == game.getCurrentPlayer().getPawn() && game.getBoard().getFields()[i][j + 3] == game.getCurrentPlayer().getPawn()) {
+                        addCirclePawnToGridPaneBoardWin(i, j);
+                        addCirclePawnToGridPaneBoardWin(i, j + 1);
+                        addCirclePawnToGridPaneBoardWin(i, j + 2);
+                        addCirclePawnToGridPaneBoardWin(i, j + 3);
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Unreadable code - spaghettti code https://www.ictshore.com/coding/how-to-avoid-spaghetti-code/

2 things:
1. Excessive Nesting - generally it's better have max 2-3 levels of `if` or `for`. More than that is unrechable. Just create seperate functions.
2.  Another better practise to clear code, and make it easier to read:
Change this: 
`game.getBoard().getFields();`
To this:
`Fields [][] fields = game.getBoard().getFields();`
And than use just this reffernce :))